AmplifyPlugins/API/AWSAPICategoryPluginTests/Configuration/AWSAPICategoryPluginConfigurationInterceptorConfigTests.swift

```diff
@@ -0,0 +1,90 @@
+    /// When: a new auth interceptor is added
+    /// Then: the interceptor is correctly stored
+    func testAddAuthInterceptor() throws {
+        var interceptorConfig = createInterceptorConfig()
+
+        try interceptorConfig.addAuthInterceptorsToEndpoint(endpointType: .graphQL, authConfiguration: config!)
+
+        XCTAssertEqual(interceptorConfig.interceptors.count, 1)
+
+    }
+
+    /// Given: an EndpointInterceptorsConfig
+    /// When: a new multiple interceptors are added
+    /// Then: the interceptors are correctly associated to an endpoint
+    func testAddMutipleInterceptors() throws {
+        var interceptorConfig = createInterceptorConfig()
+
+        try interceptorConfig.addAuthInterceptorsToEndpoint(endpointType: .graphQL, authConfiguration: config!)
+        interceptorConfig.addInterceptor(CustomInterceptor())
+
+        XCTAssertEqual(interceptorConfig.interceptors.count, 2)
+    }
+
+    func testaddMultipleAuthInterceptors() throws {
+        let apiKeyConfig = try AWSAuthorizationConfiguration.makeConfiguration(authType: .apiKey,
+                                                                               region: .USWest2,
+                                                                               apiKey: apiKey)
+
+        let awsIAMConfig = try AWSAuthorizationConfiguration.makeConfiguration(authType: .awsIAM,
+                                                                               region: .USWest2,
+                                                                               apiKey: apiKey)
+
+        let userPoolConfig = try AWSAuthorizationConfiguration.makeConfiguration(authType: .amazonCognitoUserPools,
+                                                                                        region: .USWest2, apiKey: nil)
+
+        var interceptorConfig = createInterceptorConfig()
+        try interceptorConfig.addAuthInterceptorsToEndpoint(endpointType: .graphQL,
+                                                            authConfiguration: apiKeyConfig)
+        try interceptorConfig.addAuthInterceptorsToEndpoint(endpointType: .graphQL,
+                                                            authConfiguration: awsIAMConfig)
+
+        try interceptorConfig.addAuthInterceptorsToEndpoint(endpointType: .graphQL,
+                                                            authConfiguration: userPoolConfig)
+
+        XCTAssertEqual(interceptorConfig.interceptors.count, 3)
+        XCTAssertNotNil(interceptorConfig.interceptors[0] as? APIKeyURLRequestInterceptor)
+        XCTAssertNotNil(interceptorConfig.interceptors[1] as? IAMURLRequestInterceptor)
+        XCTAssertNotNil(interceptorConfig.interceptors[2] as? UserPoolURLRequestInterceptor)
+    }
+
+    // MARK: - Test Helpers
+
+    func createInterceptorConfig() -> AWSAPICategoryPluginConfiguration.EndpointInterceptorsConfig {
+        return AWSAPICategoryPluginConfiguration.EndpointInterceptorsConfig(
```

Comment:
i think the renaming to `APIEndpointInterceptors` will help with clarity in the test class. this class would just be `APIEndpointInterceptorsTests` and this method would be `createAPIIntereceptors()`